src/Settings.php

```diff
@@ -0,0 +1,147 @@
+<?php /** @noinspection PhpUndefinedVariableInspection */
+
+namespace Tribe\Extensions\CustomAllEventsUrl;
+
+use Tribe__Settings_Manager;
+
+if ( ! class_exists( Settings::class ) ) {
+	/**
+	 * Do the Settings.
+	 */
+	class Settings {
+
+		/**
+		 * The Settings Helper class.
+		 *
+		 * @var Settings_Helper
+		 */
+		protected $settings_helper;
+
+		/**
+		 * The prefix for our settings keys.
+		 *
+		 * Gets set automatically from the Text Domain or can be set manually.
+		 * The prefix should not end with underscore `_`.
+		 *
+		 * @var string
+		 */
+		private $options_prefix = '';
+
+		/**
+		 * Settings constructor.
+		 *
+		 * @param $options_prefix
+		 */
+		public function __construct( $options_prefix ) {
+
+			$this->settings_helper = new Settings_Helper();
+
+			$this->set_options_prefix( $options_prefix );
+
+			add_action( 'admin_init', [ $this, 'add_settings' ] );
+		}
+
+		/**
+		 * Allow access to set the Settings Helper property.
+		 *
+		 * @see get_settings_helper()
+		 *
+		 * @param Settings_Helper $helper
+		 *
+		 * @return Settings_Helper
+		 */
+		public function set_settings_helper( Settings_Helper $helper ) {
+			$this->settings_helper = $helper;
+
+			return $this->get_settings_helper();
+		}
+
+		/**
+		 * Allow access to get the Settings Helper property.
+		 *
+		 * @see set_settings_helper()
+		 */
+		public function get_settings_helper() {
+			return $this->settings_helper;
+		}
+
+		/**
+		 * Set the options prefix to be used for this extension's settings.
+		 *
+		 * Recommended: the plugin text domain, with hyphens converted to underscores.
+		 * Is forced to end with a single underscore. All double-underscores are converted to single.
+		 *
+		 * @see get_options_prefix()
+		 *
+		 * @param string $options_prefix
+		 */
+		private function set_options_prefix( $options_prefix ) {
+			$options_prefix = $options_prefix . '_';
```

Comment:
implement this recent fix from Extension Template: https://github.com/mt-support/tribe-ext-extension-template/commit/a8051889c8f2135c32ef9ba2bc0824dd0b7823dc